staging/src/k8s.io/apiserver/pkg/endpoints/filters/request_deadline_test.go

```diff
@@ -438,6 +441,248 @@ func TestWithFailedRequestAudit(t *testing.T) {
 	}
 }
 
+func TestPerHandlerWriteTimeoutWithHTTP2(t *testing.T) {
+	t.Run("http2/write deadline set, handler never writes to ResponseWriter", func(t *testing.T) {
```

Comment:
Instead of having 3 independent t.Run() calls in this test - can you please split that into 3 separate tests?